### PR TITLE
Gwf invite validate

### PR DESCRIPTION
**Pull request by garyfoster**

### CODE REVIEW

File: app/controllers/users/invitations_controller.rb [hunk 1]
```diff
@@ -14,18 +14,11 @@ def create
       redirect_to new_user_invitation_path
       return
     elsif User.find_by_email(proposed.email)
-      if proposed.invited_first_name.blank? || proposed.invited_last_name.blank?
```
Comment (plicjo) [inline app/controllers/users/invitations_controller.rb]: This if statement was protecting against blank fields. With the removal it, and only a `flash[:success]`, what happened if there are blank fields?

File: app/controllers/users/invitations_controller.rb [hunk 2]
```diff
@@ -34,11 +27,18 @@ def create
...
       redirect_to edit_member_path(member)
     else
       respond_with_navigational(resource) { render :new }
     end
   end
+
+  private
+
+    def client_id(user)
+     return user.respond_to?(:invited_client_id) ? user.invited_client_id : nil
```
Comment (plicjo) [inline app/controllers/users/invitations_controller.rb]: Is there a user object that might not respond to `invited_client_id`?

Could you get away with the safe navigation operator? `user&.invited_client_id`? It'll return `nil` if user is nil and not blow up.

File: app/models/concerns/radbear_company.rb [hunk 3]
```diff
@@ -70,7 +70,7 @@ def member_accepted(user)
       member = self.members.where(user_id: user.id).first
 
       unless member
-        member = self.add_member(user, SecurityGroup.pending_group, user.invited_first_name, user.invited_last_name, MemberStatus.default_pending_status, nil)
+        member = self.add_member(user, user.invited_security_group, user.invited_first_name, user.invited_last_name, MemberStatus.default_pending_status, user.respond_to?(:invited_client_id) ? user.invited_client_id : nil)
```
Comment (plicjo) [inline app/models/concerns/radbear_company.rb]: What's the use-case behind the `respond_to`? What are the different things that can end up here: a user object or nil right? Are there other things?

File: app/controllers/users/invitations_controller.rb (hunk 1 above)
Comment (plicjo) [inline app/controllers/users/invitations_controller.rb]: Does the form show the validation error now? I may have written a test for this...

File: spec/dummy/config/environments/development.rb [hunk 4]
```diff
@@ -30,6 +30,8 @@
   config.action_mailer.raise_delivery_errors = false
 
   config.action_mailer.perform_caching = false
+  config.action_mailer.default_url_options = { :host => ENV.fetch("HOST_NAME") }
+  Rails.application.routes.default_url_options[:host]= ENV.fetch("HOST_NAME")
```
Comment (plicjo) [inline spec/dummy/config/environments/development.rb]: Over time, I'm a big fan of throwing a `fetch` around environment variables that are critical to the system. People get feedback right there in dev, "hey you need this"

File: app/controllers/users/invitations_controller.rb (hunk 1 above)
Comment (plicjo) [inline app/controllers/users/invitations_controller.rb]: A quick integration test for the invalid state here could confirm/preserve the old functionality https://github.com/radicalbear/groundswell/pull/393

Comment (garyfoster) [inline app/controllers/users/invitations_controller.rb]: yes

File: spec/requests/invitations_spec.rb [hunk 5]
```diff
@@ -40,16 +40,32 @@
         fill_in "Email", with: Faker::Internet.user_name + "@radbear.net"
         fill_in "First name", with: first
         fill_in "Last name", with: last
+        select "User", from: "Security group"
         click_button "Send"
         expect(page).to have_content "We invited '#{first} #{last}'"
       end
+
+      it "shows validation errors on an invalid invite" do
```
Comment (plicjo) [inline spec/requests/invitations_spec.rb]: Oh, here's the invalid test!

File: app/controllers/users/invitations_controller.rb (hunk 2 above)
Comment (garyfoster) [inline app/controllers/users/invitations_controller.rb]: not all RBMT apps have that field

File: spec/requests/invitations_spec.rb (hunk 5 above)
Comment (plicjo) [inline spec/requests/invitations_spec.rb]: But what about the empty state? When the user leaves first name and last name empty?

File: app/models/concerns/radbear_company.rb (hunk 3 above)
Comment (garyfoster) [inline app/models/concerns/radbear_company.rb]: client_id may not be in some RBMT apps

File: app/controllers/users/invitations_controller.rb (hunk 2 above)
Comment (plicjo) [inline app/controllers/users/invitations_controller.rb]: Oh ok, cool. Please disregard.

File: spec/dummy/config/environments/development.rb (hunk 4 above)
Comment (garyfoster) [inline spec/dummy/config/environments/development.rb]: ok, i'm cool w/ that if you wanna update

File: app/models/concerns/radbear_company.rb (hunk 3 above)
Comment (plicjo) [inline app/models/concerns/radbear_company.rb]: Understood. Please disregard.

File: spec/requests/invitations_spec.rb (hunk 5 above)
Comment (garyfoster) [inline spec/requests/invitations_spec.rb]: up to you if you wanna add it

File: spec/dummy/config/environments/development.rb (hunk 4 above)
Comment (plicjo) [inline spec/dummy/config/environments/development.rb]: I'm saying that it's nice we're doing this :D Nothing to update.

Comment (garyfoster) [inline spec/dummy/config/environments/development.rb]: oh, I just noticed the fetch in there, ok :)

File: app/models/concerns/radbear_company.rb (hunk 3 above)
Comment (garyfoster) [inline app/models/concerns/radbear_company.rb]: the pattern here is kind of a hack though, the other option was to override the different things in the app that needs client_id, but it was a ton of stuff and seemed silly - we may want to do something different longer term

Comment (plicjo) [inline app/models/concerns/radbear_company.rb]: Agreed and agreed lol

**Comment by plicjo** [comment]
Works on my machine!

**Pull request merged**